docs/swarm/how-swarm-mode-works/nodes.md

```diff
@@ -0,0 +1,94 @@
+<!--[metadata]>
++++
+aliases = [
+"/engine/swarm/how-swarm-mode-works/"
+]
+title = "How nodes work"
+description = "How swarm nodes work"
+keywords = ["docker, container, cluster, swarm mode, node"]
+advisory = "rc"
+[menu.main]
+identifier="how-nodes-work"
+parent="how-swarm-works"
+weight="3"
++++
+<![end-metadata]-->
+
+# How nodes work
+
+Docker Engine 1.12 introduces swarm mode that enables you to create a
+cluster of one or more Docker Engines called a swarm. A swarm consists
+of one or more nodes: physical or virtual machines running Docker
+Engine 1.12 or later in swarm mode.
+
+There are two types of nodes: [**managers**](#manager-nodes) and
+[**workers**](#worker-nodes).
+
+![Swarm mode cluster](../images/swarm-diagram.png)
+
+If you haven't already, read through the [swarm mode overview](../index.md) and [key concepts](../key-concepts.md).
+
+## Manager nodes
+
+Manager nodes handle cluster management tasks:
+
+* maintaining cluster state
+* scheduling services
+* serving swarm mode [HTTP API endpoints](../../reference/api/index.md)
+
+Using a [Raft](https://raft.github.io/raft.pdf) implementation, the
+managers maintain a consistent internal state of the entire swarm and
+all the services running on it. In most cases, one manager should be
```

Comment:
My point is, basically, that running Swarm mode with a single manager is actually
*less* HA than running a bunch of nodes without Swarm mode. If you have a single
manager, and that manager goes down (or looses connection with the workers), you
cannot do anything (sure, your containers/services keep running).

With a "non-swarm-mode" setup, you'd still be able to manually start/stop/create
containers, but since services are "managed", interacting with those containers
can be troublesome (when the manager comes back, strange things happen). Also
(but this may be fixed).

In a more serious situation, when the manager goes down *permanently*, your cluster
is a write-off; you cannot recover from that, without creating a new cluster.

We introduced Swarm mode, pointing out that there's no single point of failure,
and that's true from a design perspective (distributed store, etc.), and when
set up with multiple managers; but (unless I'm missing something), with a single
manager, the manager is the risk.

For that reason, even for a small cluster, I'm not sure we can ever recommend
running with a single manager, for anything that's "production".

I think we should point out these risks; and explain that a single manager is only
useful for testing, or other situations where loosing the swarm is no problem.